sbi/neural_nets/net_builders/vector_field_nets.py

```diff
@@ -0,0 +1,1186 @@
+        super().__init__()
+
+        self.skip_connections = skip_connections
+        self.layer_norm = layer_norm
+
+        # Input layers
+        self.input_layer_theta = nn.Linear(input_dim, hidden_features)
+        self.input_layer_x = nn.Linear(condition_dim, hidden_features)
+        self.input_merge_layer = nn.Linear(
+            hidden_features + hidden_features, hidden_features
+        )
+
+        # Time embedding
+        if time_emb_type == "sinusoidal":
+            self.time_emb = SinusoidalTimeEmbedding(
+                embed_dim=time_emb_dim, max_freq=sinusoidal_max_freq
+            )
+        elif time_emb_type == "random_fourier":
+            self.time_emb = RandomFourierTimeEmbedding(
+                embed_dim=time_emb_dim, scale=fourier_scale
+            )
+        else:
+            raise ValueError(f"Unknown time embedding type: {time_emb_type}")
+
+        # Main network layers
+        self.activation = activation()
+        self.layers = nn.ModuleList()
+        for _ in range(num_layers):
+            self.layers.append(nn.Linear(hidden_features, hidden_features))
+
+        # Layer normalization if enabled
+        if layer_norm:
+            self.layers_norm = nn.ModuleList()
+            for _ in range(num_layers):
+                self.layers_norm.append(nn.LayerNorm(hidden_features))
+
+        # Time embedding projection
+        self.time_linear_layer = nn.Linear(time_emb_dim, hidden_features)
+
+        # Output layer
+        self.output_layer = nn.Linear(hidden_features, input_dim)
+        nn.init.zeros_(self.output_layer.weight)
+
+    def forward(self, theta: Tensor, x_emb_cond: Tensor, t: Tensor) -> Tensor:
+        """Forward pass through the MLP.
+
+        Args:
+            theta: Parameters (for FMPE) or state (for NPSE).
+            x: Conditioning information.
+            t: Time parameter embedding.
+
+        Returns:
+            Vector field evaluation at the provided points.
+        """
+        # Process inputs
+        theta_emb = self.input_layer_theta(theta)
+        x_emb = self.input_layer_x(x_emb_cond)
+        h = self.input_merge_layer(
+            self.activation(torch.cat([theta_emb, x_emb], dim=-1))
+        )
+
+        # Process time embedding
+        t_emb = self.time_emb(t)
+        t_emb = self.time_linear_layer(t_emb)
+
+        # Main network forward pass
+        h = self.activation(h)
+        for i in range(len(self.layers)):
+            h_old = h
+            h = self.layers[i](h)
+            h = self.activation(h)
+            h += t_emb
+            if self.skip_connections:
+                h += h_old
+            if self.layer_norm:
+                h = self.layers_norm[i](h)
+
+        return self.output_layer(h)
+
+
+class VectorFieldAdaMLP(VectorFieldNet):
+    """MLP for vector field estimation
+
+    Architecture adapted from "Scalable Diffusion Models with Transformers"
+    (Peebles & Xie, 2022) with adaptive layer norm for conditioning.
```

Comment:
is adaMLP really from the Diffusion Models with Transformers paper?

Reply:
Actually not really sure (not sure if the MLP variant has a paper).